packages/artifact/src/internal/upload/upload-artifact.ts

```diff
@@ -44,21 +44,9 @@ export async function uploadArtifact(
 
   // get the IDs needed for the artifact creation
   const backendIds = getBackendIdsFromToken()
-  if (!backendIds.workflowRunBackendId || !backendIds.workflowJobRunBackendId) {
```

Comment:
Removed this since we'll be getting the backend ids in multiple client methods. I'll leave it up to the caller instead to handle the rejected promise. The behavior of having it resolve and requiring a check on `success` is a bit odd IMO